source/features/useful-not-found-page.tsx

```diff
@@ -29,71 +27,247 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
-		</p>,
-	);
+	// If it does, return it
+	return url.toString();
 }
 
-function init(): false | void {
-	const parts = parseCurrentURL();
-	if (parts.length <= 1 || !select.exists('[alt*="This is not the web page you are looking for"]')) {
-		return false;
+async function whatHappenedToTheFile(): Promise<Record<string, string | undefined> | void> {
+	// First, check if the file exists on the default branch
+	const urlOnDefaultBranch = await getUrlToFileOnDefaultBranch();
+	if (urlOnDefaultBranch) {
+		return {
+			type: 'isOnDefaultBranch',
+			linkToLastVersion: urlOnDefaultBranch,
+		};
 	}
 
-	const bar = <h2 className="container mt-4 text-center"/>;
+	// Else, check if the file exists in commit history
+	const lastCommitIncludingFile = await getLastCommitForFile();
+	if (lastCommitIncludingFile) {
+		return lastCommitIncludingFile;
+	}
+
+	// Else, check if the file exists in the default branch's history
+	const linkToCommitHistoryOnDefaultBranch
+		= await getLinkToCommitHistoryOnDefaultBranch();
+	if (linkToCommitHistoryOnDefaultBranch) {
+		return {
+			type: 'wasOnDefaultBranch',
+			linkToCommitHistory: linkToCommitHistoryOnDefaultBranch,
+		};
+	}
+}
+
+async function showAdditionalInformation(bar: Element): Promise<void> {
+	// Check what happened
+	const eventThatHappened = await whatHappenedToTheFile();
+	if (!eventThatHappened) {
+		return;
+	}
+
+	// Parse that and do/show something
+	if (eventThatHappened.type === 'isOnDefaultBranch') {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> only exists on the default branch.
+			</p>,
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === 'wasOnDefaultBranch') {
+		bar.after(
+			<p className="container mt-4 text-center">
+				This file used to exist on the default branch - see <a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>,
+		);
+		return;
+	}
 
+	if (eventThatHappened.type === 'removed') {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> was deleted <a href={eventThatHappened.linkToCommit}><relative-time datetime={new Date(eventThatHappened.commitDate!)}/></a> - see <a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>,
+		);
+		return;
+	}
+
+	if (eventThatHappened.type === 'renamed') {
+		bar.after(
+			<p className="container mt-4 text-center">
+				<a href={eventThatHappened.linkToLastVersion}>This file</a> was renamed to <a href={eventThatHappened.linkToNewVersion}>{eventThatHappened.newFilePath}</a> (<a href={eventThatHappened.linkToCommit}><relative-time datetime={new Date(eventThatHappened.commitDate!)}/></a>) - see <a href={eventThatHappened.linkToCommitHistory}>commit history</a>.
+			</p>,
+		);
+	}
+}
+
+async function showWhatIsMissingInPath(bar: Element): Promise<void> {
+	// Get the current URL
+	const parts = parseCurrentURL();
```

Comment:
If you add comments ensure that they are clear, _correct_, and that aren't just rephrasing the code. This one is rephrasing the code (parseCurrentURL === "get the current URL"), others are wrong or vague ("Parse that and do/show something"). If they don't add anything useful, they should not exist

Reply:
Sure, I'll go through the code and remove them if needed